### DISCUSSION TITLE
Lag dxwrapper

### DISCUSSION THREAD

**Discussion by esss625** [elishacloud/dxwrapper#323] [General]
I'm playing a Metin2 game, and I have a dll to stabilize the fps and reduce the input lag, when I enter the game my fps are 61, but the game moves laguit, the dll has dxwrapper, can you tell me why is it moving slowly and can someone help me?
[dinput8.zip](https://github.com/user-attachments/files/17938680/dinput8.zip)
This is dll. Thank you.

**Comment by elishacloud** [comment]
What settings are you using?  Have you tried enabling `Dinputto8`, `EnableDinput8Wrapper` and `FixHighFrequencyMouse`?

**Reply by esss625** [reply]
Is not my own dll, so i dont know exactly what is enabled, i need this settings legacy.flip, fps.boost and force the game to have stable fps without drops, mixed vertex proccessing and smoothness for the game, this dll stable my fps and change my visual in game, but with 62 fps is laggy. If you can check this dll and tell me what is wrong. thank you.

**Reply by esss625** [reply]
You can help me?

**Reply by elishacloud** [reply]
Can you attach the ini file you are using and your log file?

**Reply by esss625** [reply]
I think that the first dll that I sent above already directly activates some functions from dxwrapper, but I don't know which one, because I don't know how to decompile it to see what creates my lag, but I would like it if you could give me a configuration for all the features I mentioned above, and to incorporate the dll sent the first time, dinput8. or just settings for this :  legacy.flip, fps.boost and force the game to have stable fps without drops, mixed vertex proccessing and smoothness for the game, and change visual for better gameplay, (visual in metin2 helps you so that your opponents don't catch you in certain bugs, and you create them yourself)

[dxwrapper log files and ini.zip](https://github.com/user-attachments/files/17963698/dxwrapper.log.files.and.ini.zip)

**Reply by elishacloud** [reply]
You don't have any options enabled in the ini file.  Try enabling `EnableDinput8Wrapper` and `FixHighFrequencyMouse` in the ini file and then send me the log file after testing it.

**Reply by esss625** [reply]
[dxwrapper-rop2.log](https://github.com/user-attachments/files/17963734/dxwrapper-rop2.log)
I changed it, but I have a delay, and the game has a few frame drops, with the dll previously sent by me that is not made by me and I don't understand what settings it has, I have a stable fps regardless of the situation and the input lag disappears, which helps me a lot with the visuals of the game and its bugs. I need some settings for 
legacy.flip, fps.boost and force the game to have stable fps without drops, mixed vertex proccessing and smoothness for the game, and change visual for better gameplay. dinput 8 sent for the first time in the conversation has all these things, done with dxwrapper, they make my fps stable, but the game moves a little laggy and I don't know why

**Reply by elishacloud** [reply]
Try enabling the `d3d8to9` option and send me the new log file.

**Reply by esss625** [reply]
[dxwrapper-rop2.log](https://github.com/user-attachments/files/17963771/dxwrapper-rop2.log)

**Reply by elishacloud** [reply]
Looks like that did nothing.  I think this game may be using DirectX9.  So I don't really have much to help.  You could try enabling `EnableD3d9Wrapper`.  That will at least let dxwrapper hook into the game's graphics.

Then you can play around with the options under the `[d3d9]` section to see if any may help.  If you have a newer GPU you could try with `ForceDirect3D9On12`.  If you have a laptop you could try `GraphicsHybridAdapter`.  Also, `SetSwapEffectShim` may help reduce some lag.

**Reply by esss625** [reply]
And what i can toggle for stable fps at 61?

**Reply by elishacloud** [reply]
There is no toggle for a stable FPS.  You have to figure out what is causing the FPS issue and resolve it.  I gave you a few things to try.
